src/librustdoc/html/render.rs

```diff
@@ -1293,8 +1323,16 @@ impl Context {
             // pages dedicated to them.
             _ if item.name.is_some() => {
                 let dst = self.dst.join(&item_path(&item));
-                let dst = try!(File::create(&dst));
-                render(dst, self, &item, true)
+                let filename = dst.file_name().unwrap().to_str().unwrap();
+
+                let dst = try!(File::create(&dst)
+                               .map_err(|e| Error {
+                                                error: e,
+                                                filename: Some(filename.to_owned()) }));
+                match render(dst, self, &item, true) {
+                    Ok(ret) => Ok(ret),
+                    Err(e) => Err(Error { error: e, filename: Some(String::new()) }),
```

Comment:
This has the same problem as before where a blank filename is used to generate a suboptimal error message. This should just be able to use `try!`.

Also, if you don't have the `From` implementation above, how many instances of `try!` are there that no longer work? It seems like this should be one of the few sources of `io::Error` -> `Error` conversions